src/core/signer/ed25519.ts

```diff
@@ -0,0 +1,71 @@
+import { AccountAuthenticatorEd25519 } from "../../transactions/authenticator/account";
+import type { HexInput } from "../../types";
+import { Ed25519Account } from "../account";
+import { AccountAddressInput } from "../accountAddress";
+import { Ed25519PrivateKey } from "../crypto";
+import type { Signer } from "./signer";
+
+export interface Ed25519SignerConstructorArgs {
+  privateKey: Ed25519PrivateKey;
+  address?: AccountAddressInput;
+}
+
+export interface Ed25519SignerFromDerivationPathArgs {
```

Comment:
what do you think on having all crypto types in all those files under a one single `types.ts` file? same as we have under `/transactions/`

Reply:
I think it would be confusing. These are input args specifically for an Ed25519Signer for example.
I would put something under `types` if it's meant to be shared across multiple files.